Add script secrets creation

rodrigoalmeida94 opened the pull request.

chuckwondo commented on bin/create-env-secrets.sh:
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
```
Please make this file executable.

chuckwondo commented on bin/create-instance-secrets.sh:
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
```
Please make this file executable.

chuckwondo commented on bin/create-env-secrets.sh:
```diff
@@ -0,0 +1,72 @@
+    echo "This is the created secret full path: $secret_full_path"
+    echo "Description: $secret_description"
+    echo "Please input the value for $secret_name:"
+    echo "(or enter Q to skip this secret.)"
```
Perhaps we should accept an empty value as meaning "skip". If you would prefer to require a specific value, allow lowercase "q" as well. (I ran this and entered "q", and now there is a secret with a value of "q".)

rodrigoalmeida94 replied:
Good point, will add that.

chuckwondo commented on bin/create-env-secrets.sh:
```diff
@@ -0,0 +1,72 @@
+do
+    secret_name="${secret%%::*}"
+    secret_description="${secret##*::}"
+    secret_full_path="csdap-cumulus/$ENV/$secret_name"
```
We can drop the `csdap-` prefix and just use `cumulus` as the app prefix (i.e., `cumulus/$ENV/$secret_name`)

chuckwondo commented on bin/create-env-secrets.sh:
```diff
@@ -0,0 +1,72 @@
+    echo "(or enter Q to skip this secret.)"
+    echo ""
+
+    read secret_value
```
Since we are entering secrets, do we want to use the `-s` option to avoid echoing the values to the terminal?  If so, then we might need to ask the user to re-enter the secret and check that both inputs are the same, although this is likely overkill since each secret is likely copied and pasted, not manually typed out.

rodrigoalmeida94 replied:
I like that! Added it

chuckwondo commented on bin/create-env-secrets.sh:
```diff
@@ -0,0 +1,72 @@
+    fi
+    create_command="aws secretsmanager create-secret --name '${secret_full_path}' 
+    --description '${secret_description}' --secret-string '$secret_value'"
+    echo $create_command
```
We probably don't want to echo the command so that the secret doesn't appear in the output.

rodrigoalmeida94 replied:
Right... removed!

chuckwondo commented on bin/create-env-secrets.sh:
```diff
@@ -0,0 +1,72 @@
+secrets_description=(
+    "csdap-client-id::The CSDAP client id."
+    "csdap-client-password::The CSDAP client password."
+    "csdap-host-url::The CSDAP host url."
```
The URL doesn't need to be a secret. It will be an env var.

chuckwondo commented on bin/create-instance-secrets.sh:
```diff
@@ -0,0 +1,57 @@
+    "cmr-password::Earthdata Login (EDL) credentials, password."
+    "urs-client-id::Earthdata application client ID."
+    "urs-client-password::Earthdata application password for authentication"
+    "api-users::Comma-separated list of double-quoted Earthdata Login usernames of authorized users."
```
We can leave out `api-users`.

chuckwondo reviewed (changes requested):
I like the scripts.  I learned something new in bash scripting. I wasn't aware of `select in`.

Most of my comments on the "env" script also apply to the "instance" script.

At a later point (or perhaps now), we may want to add the ability to update secrets because if I run the scripts a second time, I cannot replace the secrets.  I would just get errors indicating the secrets already exist.

That is, instead of just using the create-secret command, first use get-secret-value (or describe-secret) to see if you get an error.  If so (secrete doesn't exist), use create-secret. Otherwise, use put-secret-value.

rodrigoalmeida94 commented:
Great idea @chuckwondo ! Added that functionality as you described - if the secret exists, update the value, if not, create a secret. Also implemented the other suggestions.
